server/src/main/kotlin/suwayomi/tachidesk/graphql/server/primitives/DurationAsString.kt

```diff
@@ -0,0 +1,122 @@
+package suwayomi.tachidesk.graphql.server.primitives
+
+import graphql.GraphQLContext
+import graphql.execution.CoercedVariables
+import graphql.language.StringValue
+import graphql.language.Value
+import graphql.scalar.CoercingUtil
+import graphql.schema.Coercing
+import graphql.schema.CoercingParseLiteralException
+import graphql.schema.CoercingParseValueException
+import graphql.schema.CoercingSerializeException
+import graphql.schema.GraphQLScalarType
+import java.util.Locale
+import kotlin.time.Duration
+
+val GraphQLDurationAsString: GraphQLScalarType =
+    GraphQLScalarType
+        .newScalar()
+        .name("Duration")
+        .description("An ISO-8601 encoded duration string")
+        .coercing(GraphqlDurationAsStringCoercing())
+        .build()
+
+private class GraphqlDurationAsStringCoercing : Coercing<Duration, String> {
+    private fun toStringImpl(input: Any): String =
+        when (input) {
+            is Duration -> input.toIsoString()
+            is String -> Duration.parse(input).toIsoString()
```

Comment:
Everywhere else we use Duration.toString(), so its probably best to standardize on that. Unless there is a drawback to that.

```suggestion
            is Duration -> input.toString()
            is String -> Duration.parse(input).toString()
```